src/services/codefixes/fixAddMissingNewOperator.ts

```diff
@@ -0,0 +1,29 @@
+/* @internal */
+namespace ts.codefix {
+    const fixId = "addMissingNewOperator";
+    const errorCodes = [Diagnostics.Value_of_type_0_is_not_callable_Did_you_mean_to_include_new.code];
+    registerCodeFix({
+        errorCodes,
+        getCodeActions(context) {
+            const { sourceFile, span } = context;
+            const identifierWithoutNew = getIdentifier(sourceFile, span.start);
+            const changes = textChanges.ChangeTracker.with(context, t => addMissingNewOperator(t, sourceFile, identifierWithoutNew));
+            return [createCodeFixAction(fixId, changes, Diagnostics.Add_missing_new_operator_to_caller, fixId, Diagnostics.Add_missing_new_operator_to_all_callers)];
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) =>
+            addMissingNewOperator(changes, context.sourceFile, getIdentifier(diag.file, diag.start))),
+    });
+
+    function getIdentifier(sourceFile: SourceFile, pos: number): Identifier {
+        const token = getTokenAtPosition(sourceFile, pos);
+        Debug.assert(token.kind === SyntaxKind.Identifier);
+        Debug.assert(isCallExpression(token.parent));
+        return <Identifier>token;
+    }
+
+    function addMissingNewOperator(changes: textChanges.ChangeTracker, sourceFile: SourceFile, identifierWithoutNew: Identifier): void {
+        const newTypeNode = createNew(identifierWithoutNew, /*typeArguments*/ undefined, /*argumentsArray*/ undefined);
+        changes.replaceNode(sourceFile, identifierWithoutNew, newTypeNode);

```

Comment:
I am so surprised that this is working. You should be replacing the `CallExpression`, not the identifier. The fact that this works is likely a bug.

Reply:
I think it works because the identifier doesn't have the `typeArguments` and the `arguments` themselves here and therefore the new can replace it properly.

I guess it only works in the "simple" scenarios that I added, and indeed in the test without the identifier it doesn't.
Could you please advise on a better approach in which I use the `CallExpression`?
BTW, the `CallExpression` doesn't have `typeArguments` nor `arguments`, so how can I pass the original ones as you mentioned?

Thanks